File: API/temsAPI/temsAPI/Helpers/Filters/LogFilter.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;
using temsAPI.Helpers.Filters.Contracts;

namespace temsAPI.Helpers.Filters
{
    public class LogFilter : IPaginationFilter, IEquipmentLabel
    {
        public int Skip { get; set; } = 0;
        public int Take { get; set; } = int.MaxValue;

        public string EquipmentId { get; set; }
        public string RoomId { get; set; }
        public string PersonnelId { get; set; }
        public string SearchValue { get; set; } // for the search bar (if any) (support might be added later)
        public List<string> IncludeLabels { get; set; }
    }
}

File: Old tems/UI/temsUI/src/app/helpers/filters/log.filter.ts
import { IsNullOrUndefined } from "../validators/validations";

export class LogFilter {
    skip: number = 0;
    take: number = 2_147_483_647; // C# Int.MaxValue
    // Provide value only for the entity whose logs should be fetches.
    // Leave the rest as null / undefined.
    equipmentId: string;
    roomId: string;
    personnelId: string;
    // When fetching room and personnel logs, user can choose between including equipment-related logs or not.
    // and if so, logs of which type of equipment to be included.
    includeLabels: string[]; // Equipment, Component, Part
    searchValue: string; // for the search bar (if any)

    isValid = () => !(IsNullOrUndefined(this.equipmentId) && IsNullOrUndefined(this.roomId) && IsNullOrUndefined(this.personnelId)) 
}